modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/MethodArgumentResolverDelegate.java

```diff
@@ -246,6 +249,8 @@ private void initArgumentResolvers() {
       Object parameterValue = valueSupplier.get();
       if (parameterValue == null) {
         return resolvePrimitiveTypeDefaultValue(parameterType);
+      } else if (parameterType.equals(InputStream.class) && parameterValue instanceof CursorStream) {
```

Comment:
should this wrapping be done for other InputStream impls that are not ours?

Reply:
I would assume that we are the only ones implementing CursorStream, do you think of a valid case where someone else is implementing it?